Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -201,6 +227,22 @@ public InteractiveBrokersBrokerage(IAlgorithm algorithm, IOrderProvider orderPro
             _clientId = IncrementClientId();
             _agentDescription = agentDescription;
 
+            Log.Trace("InteractiveBrokersBrokerage.InteractiveBrokersBrokerage(): Starting IB Automater...");
+
+            // start IB Gateway
+            _ibAutomater = new IBAutomater.IBAutomater(twsDirectory, ibVersion, userName, password, tradingMode, port);
+            _ibAutomater.OutputDataReceived += OnIbAutomaterOutputDataReceived;
+            _ibAutomater.ErrorDataReceived += OnIbAutomaterErrorDataReceived;
+            _ibAutomater.Exited += OnIbAutomaterExited;
+            _ibAutomater.Start(false);
+
+            // wait a few seconds for IB to start up
+            _ibAutomaterInitializeEvent.WaitOne(TimeSpan.FromSeconds(30));
```

Comment:
Wondering what happens if it timed out and there were no errors, `CheckIbAutomaterErrors()` didn't throw?

Reply:
The 30-second wait was previously required for TWS (which had a longer startup time than IB Gateway), now we don't support TWS anymore so this timeout could be shortened.
In this PR I added the initialize event to catch the two handled errors (existing session and security dialog) immediately, without waiting the full 30 seconds. 
Currently, the reset event is only set in those two cases: maybe I could try setting the event upon successful initialization.

Reply:
> maybe I could try setting the event upon successful initialization.

That would be awesome. Also I think we should handle somehow the case: that if past the 30 seconds it didn't connect and we didn't detect expected errors?

Reply:
Actually, the real connection is not done by the IBAutomater (it only inputs credentials for login and changes a few settings in the IB Gateway GUI). The 30 seconds delay has always been there only to give enough time for the TWS/IBG process to be loaded and ready.
The real connection to the IB C# API is done later on in the `Connect` method.

For IBAutomater, other things that could go wrong are unhandled dialogs/popups, which will now be much easier to notice and fix, since now every window event (opened, closed, activated) is now logged in the LEAN syslog.